### DISCUSSION TITLE
Hover values on android

### DISCUSSION THREAD

**Discussion by dorianhh** [dbuezas/lovelace-plotly-graph-card#496] [Q&A] [answered]
If i run homeassistant on a computer browser, i see the the values of the curves under the the mouse cursor.
On a android touchscreen (homeassistant app and tried different browsers) i cant get the values to be shown.
If i press several times on the same place, some times for a very short time the values appear (but maybe 0.3 seconds).

How can i enable the feature to show the values when pressing on the curve?

Best regards

**Comment by dbuezas** [comment, marked as answer]
Unfortunately a limitation of current plotly js. I hope they improve that soon, there's not much i can do from the card

**Reply by dorianhh** [reply]
Hello David, ok i understand. I thought maybe it was a wrong configuration on my side. Thanks so much for the card, i really love it.

**Reply by dbuezas** [reply]
Happy to hear!
Btw, I may ve wrong. I know this used to be an issue, but it looks like they fixed it and the way to get the hover tooltip in mobile is to double tap and keep pressed the second time:
Press, release, press and keep pressed.

Cool, I didn't know this myself!

**Reply by dorianhh** [reply]
Thank you very much for the update. Just tried it and you are right. Holding was the magic.
Is there a way to change the alignment of the popup text? Right now its always towards the finger, i would like to align it at least to the other side, so its easier to read the value.
Or maybe that the popup stays for 5 seconds after releasing? Then there would be enough time to read the value.

**Reply by dbuezas** [reply]
Testing a bit with plotlyjs outside of this card, i see it works with a single tap, and doesn't go away when one lifts the finger. Maybe this card breaks that somehow.
Anyway, regarding moving it, i don't think it's possible (except for left/top alignment, which would still be included by your finger

**Reply by dbuezas** [reply]
Ok, found what the issue is. The coordinates are wrong somehow. If you tap once but like half a lot above and the left, the hover tooltip appears.

**Comment by jdaleo23** [comment]
I know this has already been answered, but I was just wondering if there was any progress (from the plotly js side) regarding mobile usage?  I replaced my apexcharts with plotly because it's easier to work with with multiple yaxis and overall customization, but the mobile experience on the apexcharts is similar to the desktop browser use (mouse over / finger over) vs double tap and hold (which can be a bit clunky, especially at bars all the way to the left / right and inconsistently remaining on screen).  Just wondering if there has been any workaround, or modifications that can be done a bit deeper down to make it easier.

**Reply by dbuezas** [reply]
I think the issue is that plotly doesn't like being used inside a shadow DOM element: the 0,0 coordinate for hovering is not on the plot but on the whole page. There isn't any update unfortunately.
